examples/tutorials/text.py

```diff
@@ -0,0 +1,128 @@
+:meth:`pygmt.Figure.text`.
+"""
+
+import pygmt
+
+########################################################################################
+# Basic map annotation
+# --------------------
+#
+# Text annotations can be added to a map using the :meth:`text` method of the
+# :class:`pygmt.Figure`.
+#
+# Full details of the GMT6 command `text` can be found `here<https://docs.generic-mapping-tools.org/latest/text.html>`_.
+# The Python binding to this command is documented `here<https://www.pygmt.org/latest/api/generated/pygmt.Figure.text.html>`_.
+#
+# Here we create a simple map and add an annotation using the ``text``, ``x``,
+# and ``y`` arguments to specify the annotation text and position in the
+# projection frame. ``text``, ``x``, and ``y`` accept `int`, `str`, or `float`.
+
+fig = pygmt.Figure()
+with pygmt.config(MAP_FRAME_TYPE="plain"):
+    fig.basemap(region=[108, 120, -5, 8], projection="M20c", frame="a")
+fig.coast(land="black", water="skyblue")
+
+# Plotting text annotations using single elements
+fig.text(text="SOUTH CHINA SEA", x=112, y=6)
+
+# Plotting text annotations using lists of elements
+fig.text(text=["CELEBES SEA", "JAVA SEA"], x=[119, 112], y=[3.25, -4.6])
+
+fig.show()
+
+########################################################################################
+# Changing font style
+# -------------------
+# The size, family/weight, and colour of an annotation can be specified using the ``font`` argument.
+#
+# A list of all recognised fonts can be found `here<https://docs.generic-mapping-tools.org/latest/cookbook/postscript_fonts.html>`_),
+# including details of how to use non-default fonts.
+
+fig = pygmt.Figure()
+with pygmt.config(MAP_FRAME_TYPE="plain"):
+    fig.basemap(region=[108, 120, -5, 8], projection="M20c", frame="a")
+fig.coast(land="black", water="skyblue")
+
+# Customising the font style
+fig.text(text="BORNEO", x="114.", y=0.5, font="22p,Helvetica-Bold,white")
+
+fig.show()
+
+########################################################################################
+# Plotting from a text file
+# -------------------------
+#
+# It is also possible to add annotations from a file containing `x`, `y`, and
+# `text` fields. Here we give a complete example.
+
+fig = pygmt.Figure()
+with pygmt.config(MAP_FRAME_TYPE="plain"):
+    fig.basemap(region=[108, 120, -5, 8], projection="M20c", frame="a")
+fig.coast(land="black", water="skyblue")
+
+# Plot region names / sea names
+fig.text(textfiles="examples.txt", font="22p,Helvetica-Bold,white")
+
+# Plot names of seas
+fig.text(text=["CELEBES SEA", "JAVA SEA"], x=[119, 112], y=[3.25, -4.6])
+fig.text(text="SULU SEA", x=119.12, y=7.25, angle=-40)
+fig.text(text="SOUTH CHINA SEA", x=112, y=6, angle=40)
+fig.text(text="MAKASSAR STRAIT", x=118.4, y=-1, angle=65)
+fig.show()
+
+########################################################################################
+# ``justify`` argument
+# --------------------
```

Comment:
I really like this `justify` plot. It would be nice to put it in the gallery since there are other GMT commands that use it (e.g. legend), but let's leave it here for now.

Reply:
See also https://docs.generic-mapping-tools.org/latest/cookbook/features.html#placement-of-text (maybe link to it?).